src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectConfigurationExtensions.cs

```diff
@@ -52,5 +54,51 @@ internal static bool EqualIgnoringTargetFramework(this ProjectConfiguration proj
 
             return true;
         }
+
+        /// <summary>
+        /// Produces a string containing each dimension's value, in idiomatic order (configuration, platform, then any targets),
+        /// such as <c>Debug|AnyCPU|net7.0</c>.
+        /// </summary>
+        /// <remarks>
+        /// Calling <c>ToString</c> on <see cref="ProjectConfiguration"/> excludes the target framework from the first configuration,
+        /// meaning it is not suitable when the full set of dimensions is needed.
+        /// </remarks>
+        /// <param name="projectConfiguration"></param>
+        /// <returns></returns>
+        internal static string GetDisplayString(this ProjectConfiguration projectConfiguration)
```

Comment:
This looks like a perfect method to have some unit tests for. It has a bunch of different conditions to appending elements to the result string. Also, the unit test acts as pseudo-documentation on what the different formats of the strings look like.

Reply:
I added test cases.